packages/alfa-dom/src/matches.ts

```diff
@@ -429,22 +430,19 @@ function matchesPseudoClass(
       return options.scope === element;
 
     // https://drafts.csswg.org/css-scoping/#host-selector
-    case "host":
+    case "host": {
       // Do not allow prefix (e.g. "div:host")
       if (root !== selector) {
         return false;
       }
 
-      if (
-        options.treeContext === undefined ||
-        !isShadowRoot(options.treeContext)
-      ) {
+      const treeCtx = options.treeContext;
```

Comment:
I'm reminded a bit of the whole Golang `ctx context.Context` "stutter" when reading this 🙈 This should stutter a little less:

```ts
const { treeContext } = options;
```

Reply:
Sure. 😁 I was trying to avoid the Prettier formatting kicking in by keeping it short.

I guess I could also do:
```const { treeContext: treeCtx } = options;```
but I'll go for the object destructuring to avoid the redundancy.